### DISCUSSION TITLE
Springframework time out error

### DISCUSSION THREAD

**Discussion by kkamran1010** [SecurityRiskAdvisors/VECTR#282] [Q&A]
I am on an older version of VECTR still 8.9.4 due to my deployment being outside the normal. 

i am running into this error recently and the webserver will not start. Not sure if the old images maybe got removed or someone has updated a package on my server and broken the deployment.

My Mongo server is a remote server, i can get into mongo using shell and see the databases etc. But i am getting the below error in the docker compose output and the webui does not come up.

`com.mongodb.MongoTimeoutException: Timed out after 30000 ms while waiting for a server that matches ReadPreferenceServerSelector{readPreference=primary}. Client view of cluster state is {type=UNKNOWN, servers=[{address=xx.xx.xx.xx:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {java.net.SocketTimeoutException: Connect timed out}}]
`

**Comment by SRAPSpencer** [comment]
This is likely an issue on your docker host with the host firewall or other connectivity being blocked from the docker network to the remote. We often see this when packages are updated on the docker host. We don't offer support for remote databases with community edition, but other community members are welcome to comment if they would like.

**Comment by p-b--** [comment]
Migrating to discussion.

**Comment by kkamran1010** [comment]
i can telnet over the port 27017 from the docker host. Not sure if that is a good enough test to confirm there is no network issue between the docker host and the remote mongo server.
